offchain/dispatcher/src/dispatcher.rs

```diff
@@ -169,15 +140,19 @@ async fn process_block(
     broker: &(impl BrokerSend + BrokerReceive),
 
     claim_sender: ClaimSender,
-) -> Result<ClaimSender> {
+) -> Result<ClaimSender, DispatcherError> {
     trace!("Querying rollup state");
-    let state = client.query_state(initial_state, block.hash).await?;
+    let state = client
+        .query_state(initial_state, block.hash)
```

Comment:
Not from this PR exactly, but can you rename `client` to `state_server` since you changed things uptop? Hehehehe.